### PR TITLE
update sorting

### PR DESCRIPTION
**Pull request by AntoniosMalak**

### CODE REVIEW

**Review by AntoniosMalak — COMMENTED** [review]
update

**Comment by AntoniosMalak** [comment]
What should I do else?
@ivy-leaves @JamieLine

**Comment by JamieLine** [comment]
Hi! Thanks for the PR!

I've had a quick look through, and I have some questions / concerns. If you disagree with / would like to question anything I say here, please feel free. If I mention that something happens in a specific function, and it needs to be changed, please perform that change in the other affected functions if needed. Remember that just because I am asking why a change is made doesn't necessarily mean it has to be reverted, it just means I am struggling to see what the rationale behind the change is.

1. The docstring for `argsort` links to `tan` rather than `argsort`. This should be reverted to link to `argsort`.
2. In the docstring for `argsort`, there is a `:code:` tag that is incomplete (it says `code:` and is missing the first `:`)
3. The header in the docstring for `argsort` says `Functional Examples` when it should say `Examples`.
4. Why were some examples removed from `argsort`?
5. Why were the calls to the backend `argsort`, `sort`, and `searchsorted` functions changed to call the backend with a positional argument rather than a keyword argument? In my tests, the functions don't run correctly with this set-up, instead they raise a `TypeError: argsort() takes 1 positional argument but 4 positional arguments (and 1 keyword-only argument) were given` instead.
6. Why have the `handle_array_like` and `handle_exceptions` decorators been removed?
7. Why has `sort` had almost its entire docstring removed?
8. Why has the `# Extra #` comment heading been removed? It was after `sort`, not in a docstring but in between functions.
9. In `searchsorted`, the argument `ret_dtype` has been given a type-hint of `int` but the default value is `ivy.int64` which is an `ivy.IntDtype` object. The type hint should probably be either an `ivy.IntDtype` or a parent class of that type depending on which is more appropriate.
10. In `sort`s docstring, it says "with the same dtype and shape as values" but `values` isn't defined, it should be `x`.
11. The blank lines between the examples for `sort` should be restored, removing those lines breaks the tests for the examples as the test will be unable to tell where each example starts and ends.

If you have any questions or comments please let me know!

**Comment by AntoniosMalak** [comment]
Hi, @JamieLine thanks for your comments!
Can you check it now I changed the code as you mentioned?

**Comment by AntoniosMalak** [comment]
Hi, @ivy-seed can you check it know?

**Comment by JamieLine** [comment]
Hi! I cannot review it as I no longer work for Ivy, all on good terms I just need to focus on my studies. @CatB1t will review it in good time, I believe the standard response time is up to one week.

**Comment by AntoniosMalak** [comment]
Okay, and Thanks for your time and previous comments and I hope you will do great in your study!

**Comment by AntoniosMalak** [comment]
Hi, @CatB1t  may you review it.

**Comment by CatB1t** [comment]
Hello @AntoniosMalak , can you please reference the issue number?

**Comment by AntoniosMalak** [comment]
Yeah, sure. Here it is #7116

**Comment by AntoniosMalak** [comment]
Hi, @CatB1t may you check my code to merge or close pr or make an update in code, it's been a week ago!

**Comment by CatB1t** [comment]
Hello @AntoniosMalak , your issue doesn't seem to reference a task form the ToDo list, can you please make sure your issue is referenced in the ToDo list, so I can check which task you're trying to complete? Thanks!

**Comment by AntoniosMalak** [comment]
This is Reformat sorting #779 in ToDo list.

**Comment by AntoniosMalak** [comment]
Hi, @CatB1t Can you review it now?

**Comment by AntoniosMalak** [comment]
Hi, @CatB1t Can you review it, pleases? it has been 10 days without any review so may you check it to close this pr or edit the code!

File: ivy/functional/ivy/sorting.py [hunk 1]
```diff
@@ -254,11 +252,10 @@ def sort(
 def searchsorted(
     x: Union[ivy.Array, ivy.NativeArray],
     v: Union[ivy.Array, ivy.NativeArray],
-    /,
+    side: str = "left",
```
Comment (CatB1t) [inline ivy/functional/ivy/sorting.py:255]: Why exactly are we moving these to be positional only? we should follow the Array API standard, but given that they don't have `searchsorted` in their standard, we follow the one that is most similar, in our case `argsort` and `sort`, therefore we don't need to do these changes.

**Review by CatB1t — CHANGES_REQUESTED** [review]
Hello @AntoniosMalak, you seem to be making multiple changes which isn't related to the issue, can you revert those?

**Comment by AntoniosMalak** [comment]
Hi @CatB1t, I did these changes as mentioned in this [comment](https://github.com/unifyai/ivy/pull/7119#issuecomment-1318855395), but I revert these things you mentioned can you check it now?

**Comment by AntoniosMalak** [comment]
Hi, @CatB1 may u review it, please?

**Comment by AntoniosMalak** [comment]
@CatB1t a week passed!

File: ivy/functional/ivy/sorting.py [hunk 2]
```diff
@@ -321,4 +320,5 @@ def searchsorted(
         sorter=sorter,
         out=out,
         ret_dtype=ret_dtype,
+        x, v, side=side, sorter=sorter, out=out, ret_dtype=ret_dtype,
```
Comment (CatB1t) [inline ivy/functional/ivy/sorting.py:323]: These are already passed in, could you revert that?

File: ivy/functional/ivy/sorting.py [hunk 3]
```diff
@@ -138,12 +138,12 @@ def argsort(
 @handle_array_like
 def sort(
     x: Union[ivy.Array, ivy.NativeArray],
+    axis: int = -1,
```
Comment (CatB1t) [inline ivy/functional/ivy/sorting.py:141]: According to the array API standard, this is not correct. look here: https://data-apis.org/array-api/latest/API_specification/generated/signatures.sorting_functions.sort.html

**Review by CatB1t — CHANGES_REQUESTED** [review]
Also could you revert changes that are not related to your issue #7116 ? 
on a final note, seems like the function is already reformatted, but that wasn't marked properly on the ToDo list! we apologize for that, could you please pick a up another task to work on? 😅

**Comment by AntoniosMalak** [comment]
@CatB1t So I should pick another one after 3 weeks ago :( 
Okay should I close this pr or just pick another one?

**Pull request closed without merging**